src/components/structures/MatrixChat.tsx

```diff
@@ -1206,6 +1208,30 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
     private async onLoggedIn() {
         ThemeController.isLogin = false;
         this.themeWatcher.recheck();
+        StorageManager.tryPersistStorage();
+
+        if (SettingsStore.isLevelSupported(SettingLevel.ACCOUNT) &&
+            MatrixClientPeg.currentUserIsJustRegistered() &&
+            SettingsStore.getValue("ftueUseCaseSelection") === null) {
+            this.setStateForNewView({ view: Views.USE_CASE_SELECTION });
+
+            SettingsStore.watchSetting("ftueUseCaseSelection", null,
+                (originalSettingName, changedInRoomId, atLevel, newValueAtLevel, newValue) => {
+                    if (newValue !== null && this.state.view === Views.USE_CASE_SELECTION) {
+                        this.onShowPostLoginScreen();
```

Comment:
we should also clean up the watcher here, but instead can we use the dispatcher? The settings store isn't really meant for this ephemeral signaling type stuff.

Reply:
I’ll refer to the comments on the posthog analytics toast (which is happening after this code runs), which explains why one should use a watcher here:

https://github.com/matrix-org/matrix-react-sdk/blob/8bb1c7567f69f8c5b7a79876a3d6848af61d17bc/src/components/structures/MatrixChat.tsx#L1290-L1305

Reply:
we need a comment to refer to that comment then, please :)